From 2e4d560726541263ca688c79d3f9c7294ae5f0d4 Mon Sep 17 00:00:00 2001
From: Jeff Sharkey <jsharkey@android.com>
Date: Fri, 9 Feb 2018 10:06:59 -0700
Subject: [PATCH 122/306] Shore up dividing by zero protections.

We're already trying to protect against zero elsewhere in this file,
but we've seen evidence of it still happening, so add last-ditch
protections.

Test: builds, boots
Bug: 73060623
Change-Id: I0549ed08f906521a8dbd9ff6162f80fa9b388434
(cherry picked from commit a7be8da9adefaf6d55f7c90b95d0c6f9e66fca63)
---
 .../core/java/com/android/server/net/NetworkStatsCollection.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/net/NetworkStatsCollection.java b/services/core/java/com/android/server/net/NetworkStatsCollection.java
index 961a4517789..3cc4d83267c 100644
--- a/services/core/java/com/android/server/net/NetworkStatsCollection.java
+++ b/services/core/java/com/android/server/net/NetworkStatsCollection.java
@@ -187,6 +187,7 @@ public class NetworkStatsCollection implements FileRotator.Reader {
      */
     @VisibleForTesting
     public static long multiplySafe(long value, long num, long den) {
+        if (den == 0) den = 1;
         long x = value;
         long y = num;
 
-- 
2.17.1

